### DISCUSSION TITLE
Clear color picker Color does not take effect

### DISCUSSION THREAD

**Discussion by a554878526** [foxglove/community#969] [Bugs]
**Description**
Click the Clear color button. The color value remains unchanged The local status is not synchronized with the external status
Before repair:

https://github.com/foxglove/studio/assets/70568691/0eca6d02-ee8f-4058-bfab-eb1ab254c326

After restoration:

https://github.com/foxglove/studio/assets/70568691/958b9fa1-96da-4d4e-afd9-bc4a02f3cd54

Repair scheme:
![image](https://github.com/foxglove/studio/assets/70568691/8e733f22-4b76-4ef4-a1f7-6a6942817e58)

- Version:
- Platform:
- Data source (e.g. bag file, mcap file, rosbridge, ROS 1/2 native):

**Steps To Reproduce**

**Expected Behavior**

**Comment by foxhubber** [comment]
Internal tracking ticket: [FG-6742](https://linear.app/foxglove/issue/FG-6742/usecolorpickercontrol-error-[foxglovestudio-7498])

**Comment by a554878526** [comment]
I want to submit pr but failed at [https://github.com/foxglove/studio/pull/7499](url)

**Comment by a554878526** [comment]
How can I become a contributor

**Comment by a554878526** [comment]
![image](https://github.com/foxglove/studio/assets/70568691/f41232c4-9a12-41e8-8d0d-b3b7011ef90a)
Why was this item not adopted

**Comment by indirectlylit** [comment]
Hi @a554878526, thank you for this report. I would be happy to try and help.

In your first video, I see that the color goes blue → pink → yellow. Had you set the color from yellow to blue before the video started recording?

This might be behaving as intended because yellow is the default color, and the "clear" button action is "reset to default". Perhaps you were expecting it to behave like "undo"?

thank you for helping me understand!

----

Internal tracking ticket: [FG-6742](https://linear.app/foxglove/issue/FG-6742/usecolorpickercontrol-error-[foxglovestudio-7498])

**Comment by jtbandes** [comment]
Thanks for the report. This issue has been fixed and will be available in the next release.
